compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -2669,18 +2669,26 @@ OMR::Z::TreeEvaluator::dselectEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    TR::Register *resultReg = cg->gprClobberEvaluate(trueValueNode);
    TR::Register *conditionReg = cg->evaluate(conditionNode);
    TR::Register *falseValReg = cg->evaluate(falseValueNode);
-   if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_S390_Z13) && node->getOpCode().isDouble())
+   if ((cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_S390_Z13) && node->getOpCode().isDouble()) || (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_S390_Z14) && node->getOpCode().isFloat()))
       {
       TR::Register *vectorSelReg = cg->allocateRegister(TR_VRF);
       TR::Register *tempReg = cg->allocateRegister(TR_FPR);
       TR::Register *vzeroReg = cg->allocateRegister(TR_VRF);
-      // Convert 32 Bit register to 64 Bit (Comparison Child of the select node is 32 bit)
-      generateRRInstruction(cg, TR::InstOpCode::LLGFR, node, conditionReg, conditionReg);
+      
+      // Convert 32 Bit register to 64 Bit for Doubles (Comparison Child of the select node is 32 bit)
+      if (node->getOpCode().isDouble())
+       {
+        generateRRInstruction(cg, TR::InstOpCode::LLGFR, node, conditionReg, conditionReg);
+       }else  
+       {
```

Comment:
Please comment why we needed to left shit the condition node for float. This will help others looking into the code.

Reply:
Given the case when the node opcode is a Float Select, and the condition child is 32 bits, an **SLLG** instruction would be required to shift to the left the 32 least significant bits within the 64 bit conditionReg GPR. 
This would preserve the floating point representation of the conditionNode, as it is later moved into an FPR using **LDGR**, which would enable further floating-point operations could be performed.